packages/core/src/components/chat.ts

```diff
@@ -16,7 +16,18 @@ export interface ReceivedChatMessage extends ChatMessage {
 const encoder = new TextEncoder();
 const decoder = new TextDecoder();
 
-export function setupChat(room: Room) {
+const encode = (message: string, timestamp: number) =>
+  encoder.encode(JSON.stringify({ message, timestamp }));
+
+const decode = (message: Uint8Array) => JSON.parse(decoder.decode(message)) as ChatMessage;
+
+export function setupChat(
+  room: Room,
+  options: {
+    messageEncoder?: (message: string, timestamp: number) => Uint8Array;
```

Comment:
a couple more nits: 
could you make this accept a `ChatMessage` object directly?